New "Promotion" Mode

halelidan opened the pull request:
- Implement config switching based on promotion flag
- Modify image services to use CONFIG or PROMOTION_CONFIG as appropriate
- Skip asset deletion in promotion mode for ImageExtensionService
- Add new ImagePauseService for pausing uploaded assets

halelidan committed:
New "Promotion Mode"

- Implement config switching based on promotion flag
- Modify image services to use CONFIG or PROMOTION_CONFIG as appropriate
- Skip asset deletion in promotion mode for ImageExtensionService
- Add new ImagePauseService for pausing uploaded assets

akushnarov commented on src/config.ts:
```diff
@@ -44,11 +44,15 @@ interface Config {
   SpreadsheetApp.getActiveSpreadsheet().getSheetByName('Config')!;
-const DEFAULT_CONFIG = {
+export const promotionSheet =
+  SpreadsheetApp.getActiveSpreadsheet().getSheetByName('Promotion_Config')!;
```
I think you can make the setup even more flexible if you use in the main config sheet `Promotion Mode Config` = <PROMO_CONFIG_SHEET_NAME> (if empty it is equal to `Is Promotion Mode` = no) instead of `Is Promotion Mode` = yes|no. In this way you can have several configs for e.g. different promo seasons and quickly switch them.

halelidan replied:
I totally agree! fixed in the new commit

akushnarov commented on src/config.ts:
```diff
@@ -95,3 +100,15 @@ export const CONFIG: Config =
+      .filter(e => e[0])
+      .reduce((res, e) => {
+        return { ...res, [e[0]]: e[1] };
+      }, DEFAULT_CONFIG),
```
Instead of using the DEFAULT_CONFIG here, I would use `CONFIG`, in this way the promo config sheet can have only the fields that need to be overwritten (other config params will be copied from the main config).

halelidan replied:
same here, fixed in the new commit

halelidan committed:
Enhance promotion configuration flexibility

- Replace yes/no flag with configurable sheet name
- Make promotion config inherit from main config
- Add helper function for promotion mode checks

victor-paunescu commented on src/config.ts:
```diff
@@ -95,3 +98,22 @@ export const CONFIG: Config =
+      : {}),
+  } ?? CONFIG;
+
+export const inPromotionMode = (): boolean => {
```
Why does this need to be a function and not a constant?

victor-paunescu replied:
Nothing to change here, just curious

victor-paunescu commented on src/image-pause-service.ts:
```diff
@@ -0,0 +1,144 @@
+        ])
+        ?.items?.map(e => e.name.split('/').slice(-1)[0]);
+
+      if (uploadedImages && uploadedImages.length > 0) {
```
I think you could use optional chaining here 
```suggestion
      if (uploadedImages?.length > 0) {
```

victor-paunescu commented on src/image-pause-service.ts:
```diff
@@ -0,0 +1,144 @@
+      );
+    }
+    Logger.log('Finished Pause Process.');
+    //If script completes without timing out, clear the stored ad group ID and any triggers
```
nit: the
```suggestion
    // If the script completes without timing out, clear the stored ad group ID and any triggers
```

victor-paunescu commented on src/menu.ts:
```diff
@@ -25,6 +25,7 @@ function onOpen() {
         .addItem('Image generation', 'ImageGenerationService.manuallyRun')
         .addItem('Image upload', 'ImageUploadService.manuallyRun')
         .addItem('Image assets linking', 'ImageExtensionService.manuallyRun')
+        .addItem('Image Pause', 'ImagePauseService.manuallyRun')
```
nit: lowercase pause to be consistent
```suggestion
        .addItem('Image pause', 'ImagePauseService.manuallyRun')
```

halelidan committed:
small style fixes
